app/src/main/java/com/example/banking/di/AppModule.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.banking.di
+
+import android.app.Application
+import androidx.room.Room
+import com.example.banking.data.data_source.AccountsDatabase
+import com.example.banking.data.repository.AccountsRepository
+import com.example.banking.data.repository.AccountsRepositoryImpl
+import com.example.banking.data.repository.TransactionRepository
+import com.example.banking.data.repository.TransactionRepositoryImpl
+import com.example.banking.domain.use_case.AccountsUseCases
+import com.example.banking.domain.use_case.GetAccountsUseCase
+import com.example.banking.domain.use_case.GetCurrentAccountUseCase
+import com.example.banking.domain.use_case.GetTransactionsUseCase
+import com.example.banking.domain.use_case.InsertTransactionUseCase
+import com.example.banking.domain.use_case.TransactionsUseCases
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class AppModule {
+
+    @Provides
+    @Singleton
+    fun provideDatabase(app: Application): AccountsDatabase {
+        return Room.databaseBuilder(
+            app,
+            AccountsDatabase::class.java,
+            AccountsDatabase.DATABASE_NAME
+        ).build()
+    }
+
+    @Provides
+    @Singleton
+    fun provideAccountRepository(db: AccountsDatabase): AccountsRepository {
+        return AccountsRepositoryImpl(db.accountDao())
+    }
+
+    @Provides
+    @Singleton
```

Comment:
lifecycle of use cases is not as Singleton lifecycle because they don't save any state only invoke one unit of logic (your code is not a mistake, you can leave it as it is).